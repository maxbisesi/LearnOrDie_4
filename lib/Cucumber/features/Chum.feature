@Regression
Feature: Create, Organize, and Delete FlashCards
    This is done through the Chum and Galley tabs.
    Use the profile tab to activate modules used to add 
    speciel characters a question or answer. 

@ChumNewCard
Scenario: On the Chum tab User creates a new FlashCard
    When The client logs in as a test User

@ChumNewCardwithEquilateral
Scenario: On the Chum tab the User creates a new FlashCard with an Image of an Equilateral 
    When The client logs in as a test User

@ChumNewCardwithUnicode
Scenario: On the Chum tab the User creates a new FlashCard with Unicode characters in the Question Answer and Category
    When The client logs in as a test User

@ChumCreatedCategoryShowsInList
Scenario: When I create a new Card in a new Category, that category shows up in the Category list with the right count
    When The client logs in as a test User






