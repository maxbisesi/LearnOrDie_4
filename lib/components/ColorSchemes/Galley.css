.sets{
    position:relative;
    width: 100%;
    left: 100px;
}

.galleyTable {
    height: 900px;
    width: 100%;
    overflow-y: auto;
}

.setinprogress {
    background-color: #aaaaaa;
    border: solid 3px black;
    color: white;
    width: 100px;
    height: 25px;
    position: absolute;
    text-align: center;
    border-radius:25px;
}

.setitem{
    height: 50px;
    background-color: darkorchid;
    box-shadow: 5px 5px black;
    border-radius: 5px;
    width: 100%;
    color: white;
    font-size: 35px;
    margin-top: 10px;
    margin-bottom: 10px;
}

.longtexttd {
    width: 300px;
    max-width: 300px;
    overflow: hidden;
    height: 100px;
    max-height: 100px;
    font-size: 15px;
 }
.setitem:hover{
    cursor:pointer;
    background-color: darkmagenta;
}

.clickedset {
    background-color: orange;
    color: black;
}

.clickedset:hover {
    background-color: #ba8500;
}

.clickedquestion {
    background-color: deeppink;
    color:black;
}

.clickedquestion:hover {
    background-color: fuchsia;
}

.createset{
    overflow: auto;
    height: 400px;
    padding-bottom: 10px;
    display: grid;
    grid-template-columns: 50px 50px 50px 50px 50px;
}

.setintest{
    width:50px;
    height: 25px;
    background-image: linear-gradient(to right, #999999, #f2f2f2);
    border-radius: 8px;
    box-shadow: 3px 5px black;
    color: #000;
    font-size: 25px;
    padding-top:5px;
    padding-bottom: 5px;
    padding-left: 5px;
    padding-right: 5px;
    margin-top: 5px;
    margin-bottom: 5px;
    margin-left: 5px;
    margin-right: 5px;
    display: absolute;
}

.setinprogress:hover{
    transition: all 700ms ease-in-out;
    transform: rotate(360deg);
}

.setinfo {
    border: 1px solid deeppink;
    border-radius: 4px;
    box-shadow: 5px 10px #3333ff;
    width: 200px;
}

.setinprogressMovedown {
    transition:all 500ms ease-in;
    transform: translate(10px,30px) rotate(360deg);
}

.galleyQuestion {
    text-align: center;
    margin: 2px;
    padding: 2px;
    font-size: 20px;
    font-weight: bold;
    border: solid;
    border-color: #80bfff;
    border-width: thick;
    color: black;
    background-color: #cc33ff;
}

.galleyQuestion:hover{
    box-shadow: 0px 0px 200px #000000;
    z-index: 2;
    -webkit-transition: all 300ms ease-in;
    -webkit-transform: scale(1.5);
    -ms-transition: all 300ms ease-in;
    -ms-transform: scale(1.5);   
    -moz-transition: all 300ms ease-in;
    -moz-transform: scale(1.5);
    transition: all 300ms ease-in;
    transform: scale(1.5) translateX(50px);
}

#allquestions {
    position: absolute;
    left: 10px;
    top: 20px;
}