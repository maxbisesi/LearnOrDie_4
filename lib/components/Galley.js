import React from 'react';
import './QuestionPageSheet.css';
import PropTypes from 'prop-types';

class Galley extends React.Component {
  constructor(props, context) {
    super();
  }

  render() {
    return (
    <div id='Galley' style={{height: 1500}}>
                
        <h1 id='currentusername'>username</h1>
        {/* Card Set creation */}
        <div className='sets'>
            
            <div className='standardtable' id='allquestions' style={{height: 700}}>
            <table>
                <thead>
                <tr> 
                    <th>Card</th>
                    <th>Category</th>
                    <th>Right</th>
                    <th>Wrong</th>
                </tr>
                </thead>
                <tbody id='questionsforsets'>
                    
                </tbody>
            </table>
            </div>
            
            <div style={{position:'absolute',top: 730, left: 30}}>
                <input id='previouspage' type='button' value='<--'/>
                <input id='nextpage' type='button' value='-->'/>
            </div>
            <div style={{position:'absolute', top: 730, left: 200}}>
                <p id='pagenumber'>1</p>
            </div>
            <br/>
            <div id='setbuttons' style={{position:'absolute', top: 780, left: 30, zIndex: 2}}>
                <input id='addquestionstoset' type='button' value='Add' />
                <input id='deletequestion' type='button' value='Delete'/>
            </div>

            <div id='setinprogressid' className='setinprogress' style={{display:'none', top:880, left: 250, boxShadow: '5 5 black', zIndex: 1,height:10, width:10}}>   
            </div>
            
            <div id='setnamediv' style={{position:'absolute', top: 940, left: 130, zIndex: 4, display:'none'}}>
                <input className='setinfo' id='setname' type='text' placeholder='Name'/> <br/>
                <br/>
                <textarea className='setinfo' id='setdescription' placeholder='Set Description' ></textarea>
            </div>
            
            <div id='setbuttondiv' style={{position: 'absolute', top: 940, zIndex: 4, left: 130, display:'none'}}>
                <input id='setsavebutton' value='Save' type='button' />
            </div>
            
            <div id='sharebuttons' style={{display:'none', position:'absolute', right:20, height:45}}>
                <input id='study' type='button' value='Study'/>
                <input id='share' type='button' value='Share'/>
                <input id='remove' type='button' value='Remove'/>
            </div>
            
            <div id='activityfeed' style={{position: 'absolute', right: 20, height: 600,  width: 600, top: 50}}>
                
            </div>
            
        </div>             
    </div>
    );
  }
}
Galley.contextTypes = { user: PropTypes.object };
export default Galley;
