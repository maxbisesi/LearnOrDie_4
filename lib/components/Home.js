import React from "react";
import axios from "axios";
import "./HomeScreenCustomSheet.css";

class Home extends React.Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div id="home">
        <h1>{name}</h1>

        <input id="myprofile" type="button" value="Profile" />
        <input id="logout" name="logout" type="button" value="Logout" />
        <input id="refresh" type="button" name="name" value="Refresh" />

        <table>
          <tbody>
            <tr>
              <th>EMAIL</th>
              <th>USERNAME</th>
              <th>ID</th>
              <th>CARD COUNT</th>
              <th>RANK</th>
            </tr>
            <tr>
              <td>user email</td>
              <td id="myusername">username</td>
              <td>user id</td>
              <td>card list size</td>
              <td id="currentrank">rank</td>
              <td hidden={true} id="currentpoints">
                points
              </td>
            </tr>
          </tbody>
        </table>
        <div className="ensignia">
          <img id="ensignia" height="250px" width="150px" />
        </div>
        <br />
        <div className="standardtable">
          <table id="categorytable">
            <tbody>
              <tr id="headers">
                <th>Category</th>
                <th>Right / Wrong</th>
                <th>Card Count</th>
                <th>Filter Quesitons by Category</th>
              </tr>
              <tr>
                <td>category name</td>
                <td>category right / wrong</td>
                <td>count</td>
                <td>
                  <input
                    type="checkbox"
                    name="catFilterGroup"
                    value=" check or nah"
                    defaultChecked={false}
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <br />
        <br />
        <input type="button" id="filterbutton" value="Filter" />
        <input type="button" id="unfilterbutton" value="Remove Filter" />
        <br />
        <br />

        <p>Study History</p>
        <div id="sessionhistory">
          <table>
            <tbody>
              <tr>
                <th>Total Cards Seen</th>
                <th>Correct</th>
                <th>Incorrect</th>
                <th>Date</th>
              </tr>
              <tr>
                <td>card seen</td>
                <td>correct</td>
                <td>incorrect</td>
                <td>date</td>
              </tr>
            </tbody>
          </table>
        </div>

        {/* rank up logout popup */}
        <div id="upgradepopup" className="popup">
          <div className="popup-content">
            <div className="popup-header">
              <span className="close">&times;</span>
              <h2 id="newbadgemessage"></h2>
            </div>
            <div className="popup-body">
              <img id="popupnewbadge" height="400px" width="230px" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default Home;
