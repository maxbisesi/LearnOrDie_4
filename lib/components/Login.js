import React from 'react';

class Login extends React.Component {
    constructor(props) {
        // always
        super(props);
        this.state = {username: '', password: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        // Run after the comp output renders to the DOM
    }

    componentDidmount() {
        
    }

    handleChange(event) {
        if(event.target.name === 'username'){ this.setState({username: event.target.value}); }
        else if( event.target.name === 'password'){ this.setState({password: event.target.value}); }
    }

    async handleSubmit (e) {
        event.preventDefault();
        // axios call
    }

    render(){
        return (
        <>
            <p id='bttw'>Brave the Treacherous Waters...</p>
            <div className='centerLoginScreen'>
                <form onSubmit={this.handleSubmit}>
                    <p>username: <input name='username' type='text' value={this.state.username || ''} onChange={this.handleChange} /></p>
                    <p>password: <input name='password' type='password' value={this.state.password || ''} onChange={this.handleChange} /></p>
                    <input type='submit' value='Submit' id='loginbutton' className='' />
                </form>
                <a href='registrationPage.jsp'><button id='registerbutton' className=''>Register a new account</button></a>
            </div>
            <p id='lod'>...then Learn or Die.</p>
        </>
        );
    }
}

export default Login;