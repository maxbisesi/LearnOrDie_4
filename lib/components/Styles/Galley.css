.galley {
  position: relative;
  height: 1500px;
  width: 100%;
  background: linear-gradient(to top, #001a66, #e6ecff 70%);
}

.galley-sets {
  position: relative;
  width: 100%;
  left: 100px;
}

.galley-button {
  width: 100px;
  height: 30px;
  background-color: #da1244;
  border: none;
  color: white;
  padding: 16px 32px 16px 32px;
  text-align: center;
  font-size: 16px;
  margin: 4px 2px;
  opacity: 0.5;
  transition: 0.3s;
  display: inline-block;
  text-decoration: none;
  cursor: pointer;
}

.galley-button:hover {
  opacity: 1;
}

.galley-question_table {
  height: 1000px;
  width: 1200px;
  position: absolute;
  left: 100px;
  top: 100px;
  overflow-y: auto;
  overflow-x: visible;
}

.galley-buttons-navigation {
  position: absolute;
  top: 50px;
  left: 100px;
}

.galley-pagenumber {
  position: 'absolute';
  top: 930;
  left: 200;
}

.galley-question_table-headers { 
    border-style: solid;
    border-color: white;
    background-color: white;
    color: Blue;
}

.galley-set_table {
  width: 400px;
  padding-top: 20px;
  /* background-color: #aaaaaa; */
}

.galley-set-in_progress {
  background-color: #aaaaaa;
  border: solid 3px black;
  color: white;
  width: 100px;
  height: 25px;
  position: absolute;
  text-align: center;
  border-radius: 25px;
  display: 'none';
  top: 880;
  left: 250;
  box-shadow: '5 5 black';
  z-index: 1;
  height: 10;
  width: 10;
}

.galley-set_buttons {
}

.galley-set_item {
  height: 50px;
  background-color: darkorchid;
  box-shadow: 5px 5px black;
  border-radius: 5px;
  width: 100%;
  color: white;
  font-size: 35px;
  margin-top: 10px;
  margin-bottom: 10px;
}

.longtexttd {
  width: 300px;
  max-width: 300px;
  overflow: hidden;
  height: 100px;
  max-height: 100px;
  font-size: 15px;
}

.setitem:hover {
  cursor: pointer;
  background-color: darkmagenta;
}

.clickedset {
  background-color: orange;
  color: black;
}

.clickedset:hover {
  background-color: #ba8500;
}

.clickedquestion {
  background-color: deeppink;
  color: black;
}

.clickedquestion:hover {
  background-color: fuchsia;
}

.createset {
  overflow: auto;
  height: 400px;
  padding-bottom: 10px;
  display: grid;
  grid-template-columns: 50px 50px 50px 50px 50px;
}

.setintest {
  width: 50px;
  height: 25px;
  background-image: linear-gradient(to right, #999999, #f2f2f2);
  border-radius: 8px;
  box-shadow: 3px 5px black;
  color: #000;
  font-size: 25px;
  padding-top: 5px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  margin-top: 5px;
  margin-bottom: 5px;
  margin-left: 5px;
  margin-right: 5px;
  display: absolute;
}

.setinprogress:hover {
  transition: all 700ms ease-in-out;
  transform: rotate(360deg);
}

.setinfo {
  border: 1px solid deeppink;
  border-radius: 4px;
  box-shadow: 5px 10px #3333ff;
  width: 200px;
}

.setinprogressMovedown {
  transition: all 500ms ease-in;
  transform: translate(10px, 30px) rotate(360deg);
}

#allquestions {
  position: absolute;
  left: 10px;
  top: 20px;
}
