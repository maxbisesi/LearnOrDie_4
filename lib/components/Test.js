import React from "react";
import "./QuestionPageSheet.css";
import { connect } from "react-redux";
import { rateCard, nextCard, saveSession } from '../actions';
import Test_QandA from './Presentational/Test_QandA';

class Test extends React.Component {
  constructor(props) {
    super(props);
    console.log('Test component');
    this.state = {
      index: 0,
      rating: [],
      updates: {},
      session: {points: 0, streak: 0, rut: 0},
      showAnswer: false,
      card: this.props.card,
      answer: this.props.answer,
      category: this.props.category
    };

    this.showAnswer = this.showAnswer.bind(this);
    this.nextCard = this.nextCard.bind(this);
    this.updateCard = this.updateCard.bind(this);

  }

  nextCard(e) { 
    let result = e.target.name;

    this.setState((state,props) => { 
      // Update local state
      //state.rating.push({cardId:state.index, result:result});
      let pointsChange = this.updatePoints(result,state.session);

      // change index---------
      let ind = state.index;
      ind += 1;
      if(ind >= props.cardLength) {
        ind = 0;
      }
      this.props.next(ind);
      //-=-=-=-=-=- -=-=-=-=-=-
      
      //update cards with rating
      let cardUpdates = Object.assign({},state.updates);
      cardUpdates[this.props.id] = { [result]:this.props[result] +1 };
      let update = Object.assign({}, state, {index: ind, session: pointsChange, showAnswer: false, updates: cardUpdates} );
      
      return update;
    });

  }

  updatePoints(result,session) {
    if(result === 'nailed') {
      let newStreak = session.streak;
      let points = session.points;
      newStreak += 1;
      console.log(`new streak: ${newStreak} - old: ${session.streak}`);
      if (newStreak > 5 && newStreak < 10) { 
         points += 10; 
         return {points: points, streak: newStreak , rut: 0};
      }
      else if (newStreak > 10 && newStreak < 30) { 
        points += 20; 
         return {points: points, streak: newStreak , rut: 0};
      }
      else if (newStreak >= 30) { 
        points += 50; 
         return {points: points, streak: newStreak , rut: 0};
      } 
      points += 1;
      return {points: points, streak: newStreak , rut: 0};
    }
    else if(result === 'whiffed') {
      let missStreak = session.rut;
      let points = session.points;
      missStreak += 1;
      if (missStreak > 5 && missStreak < 10) { 
         points -= 20; 
         return {points: points, streak: 0 , rut: missStreak};
      }
      else if (missStreak > 10 && missStreak < 15) { 
        points -= 50; 
         return {points: points, streak: 0 , rut: missStreak};
      }
      else if (missStreak >= 15) { 
        points -= 150; 
         return {points: points, streak: 0 , rut: missStreak};
      } 
      points -= 10;
      return {points: points, streak: 0 , rut: missStreak};
    }
    throw new Error('updatePoints: unkown result.');
  }

  shouldComponentUpdate(nextProps, nextState) {
    if(nextState.index !== this.state.index) { return false; }
    if(nextState.session.points !== this.state.session.points){ return false; }
    return true;
  }

  componentDidMount() {
    this.setState( (state,props) => {
        return {session: props.session}
    });
  }
  componentDidUpdate(prevProps, prevState, snapshot){
    if(prevProps.card != this.props.card) {
      this.setState((state,props) => {
        return {card: this.props.card, answer: this.props.answer, category: this.props.category};
      });
    }
  }

  componentWillUnmount(){
    // update store with this session
    //this.props.rate(this.state.rating);
    console.log(this.state);
    //reset cards back to zero
    // this.props.next(0);
    // Update users points in the store, then when mounted again re instate them
    this.props.update(this.state.updates);
    this.props.saveSesh(this.state.session);
  }

  updateCard(e) { 
    // TODO animation here
    console.log('update card');
    this.setState((state,props) => {
      
      return (state.updates[this.state.id] = {
        id: this.state.id,
        card: this.state.card, 
        answer: this.state.answer, 
        category: this.state.category
      } );

    });
  }

  handleChange = ({ target }) => {
    let targ = target.name;
    if(targ === 'card'){ this.setState({ [target.name]: target.value }); }
    if(targ === 'answer'){ this.setState({ [target.name]: target.value }); }
    if(targ === 'category'){ this.setState({ [target.name]: target.value }); }
 };

  showAnswer() {
      this.setState((state,props) => { return {showAnswer: true}; } );
  }

  render() {
    return (
      <div id="test" style={{position: 'relative'}}>
        <Test_QandA handleChange={this.handleChange} card={this.state.card} answer={this.state.showAnswer ? this.state.answer : "'click show'"} category={this.state.category} />
        <div
          style={{marginTop: 10, marginBottom: 10}}
          id="setdisplay"
        ></div>

        <div id="pointsdiv">
    <p className="pointsNormal" id="points">{this.state.session.points}</p>
        </div>

        <div id="buttoncontrolgroup">
          <input type="button" name="show" onClick={this.showAnswer} value="Show"/>
          <input type="button" id="update" value="Update" onClick={this.updateCard}/>
          <input type="button" name="nailed" onClick={this.nextCard} value="Nailed it"/>
          <input type="button" name="whiffed" onClick={this.nextCard} value="Missed it"/>
          <input type="button" name="whiffed" onClick={this.previousCard} value="Previous"/>
          <input type="button" id="comebacktothisone" value="Come back to this one!"/>
          <input type="button" id="review" value="Review" />
          <p id="questinfo">
            | Rating: <span id="rating"></span> | 
            Category:{" "}<span id="cat">{this.props.category} </span> | 
            Card Number: <span id="cardid"></span> |
            Questions To Review: <span id="reviewcount"></span>|
          </p>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    card: state.currentCard.card,
    answer: state.currentCard.answer,
    whiffed: state.currentCard.whiffed,
    nailed: state.currentCard.nailed,
    id: state.currentCard.id,
    category: state.currentCard.category,
    session: state.session,
    cardLength: state.cards.length,
    session: state.session
  };
};

const mapDispatchToProps = (dispatch) => {
  return ( {
    next: (index) => dispatch(nextCard(index)),
    update: (rates) => dispatch(updateCards(rates)),
    saveSesh: (session) => dispatch(saveSession(session))
  } );
};

export default connect(mapStateToProps,mapDispatchToProps)(Test);
