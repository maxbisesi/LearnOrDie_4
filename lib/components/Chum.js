import React from "react";
import axios from "axios";
//import "./QuestionPageSheet.css";
import PropTypes from "prop-types";
import { addCardAction } from '../actions';
import { connect } from 'react-redux';
import './Styles/Chum.css';

export class Chum extends React.Component {
  constructor() {
    // TODO map state to props to pre populate category field with existing categories
    super();
    this.state = {
      card: "",
      answer: "",
      category: ""
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    console.log(`State in handleSubmit, before ${JSON.stringify(this.state)}`);
    const cardAdded = {'card': this.state.card, 'answer': this.state.answer, 'category': this.state.category};
    
    this.setState( 
      (state,props) => ({ card: "", answer: "", category: "" }),
      () => { 
        console.log(`State in handleSubmit, setState callback: ${JSON.stringify(this.state)}`);
        this.props.addCard(cardAdded); 
        console.log(`Card Added: ${JSON.stringify(cardAdded)}`);
      }
    );
  }

  handleChange(event) {
    const name = event.target.name;
    const val = event.target.value;
    // console.log(`name and val ${name} = ${val}`);
    if (name === "card") {
      this.setState((state,props) => ({ 'card': val }) );
    } else if (name === "answer") {
      this.setState((state,props) => ({ 'answer': val }) );
    } else if (name === "category") {
      this.setState((state,props) => ({ 'category': val }) );
    }
  }

  render() {
    console.log('--- Chum ---');
    return (
      <div className="chum">
        <form onSubmit={this.handleSubmit}>
          <div>
            <textarea
              value={this.state.card}
              onChange={this.handleChange}
              name="card"
              rows="15"
              cols="65"
            ></textarea>
            <br />
            <textarea
              value={this.state.answer}
              onChange={this.handleChange}
              name="answer"
              rows="15"
              cols="65"
            ></textarea>
          </div>
          <br />
          <input type="hidden" name="name" value="submit" />
          <span>Category:</span>
          <input
            value={this.state.category}
            type="text"
            name="category"
            onChange={this.handleChange}
          />
          <input type="submit" name="submit" value="submit" id="submit" />
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return ( {
    addCard: (card) => dispatch(addCardAction(card))
  } );
};
export default connect(null,mapDispatchToProps)(Chum);
