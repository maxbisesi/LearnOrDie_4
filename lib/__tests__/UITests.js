import React from 'react';
import renderer from 'react-test-renderer';
import { Chum } from '../components/Chum';
import Home from '../components/Home';
import Test from '../components/Test';
import Galley from '../components/Galley';
import Login from '../components/Login';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//Test the UI using Snapshot testing

/* describe.each
    describe.each(table)(name, fn, timeout
    describe.each([
        [1, 1, 2],
        [1, 2, 3],
        [2, 1, 3],
    ])('.add(%i, %i)', (a, b, expected) => {
    test(`returns ${expected}`, () => {
        expect(a + b).toBe(expected);
    });
*/

// create mock methods 
const [ addCard ] = new Array(1).fill(jest.fn());

function chumSetup() { 

    configure({ adapter: new Adapter() });

    const props = { 
        'addCard': addCard
    };

    const chumWrapper = shallow(<Chum {...props} />);
    return {
        props,
        chumWrapper
    };

}

describe('Test the Chum Component', () => {

  test('It renders correctly', () => {
    const {props, chumWrapper} = chumSetup();
    expect(chumWrapper).toMatchSnapshot();
  });

  test('That card and answer fields clear when submited', () => {
    const { props, chumWrapper } = chumSetup();
    //chumWrapper.find('[name="card"]').simulate('change', {target: {value:'testcard'} });
    //chumWrapper.find('[name="answer"]').simulate('change', {target: {value:'testanswer'} });
    //chumWrapper.find('textarea').first().getElement().value='testcard';
    //chumWrapper.find('textarea').last().getElement().value='testanswer';
    //expect(chumWrapper.state('card')).toBe('testcard');
    //expect(chumWrapper.state('answer')).to.toBe('testanswer');
    chumWrapper.find('input[name="submit"]').simulate('click');
    expect(chumWrapper.state('card')).toBe('');
    expect(chumWrapper.state('card')).toBe('');
  });

});


