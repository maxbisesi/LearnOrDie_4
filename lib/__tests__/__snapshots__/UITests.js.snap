// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Chum Component It renders correctly 1`] = `ShallowWrapper {}`;
